knowledge-graph/src/main/scala/io/renku/knowledgegraph/multipart/syntax.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.knowledgegraph.multipart
+
+import cats.effect.Concurrent
+import cats.syntax.all._
+import cats.{Functor, MonadThrow}
+import eu.timepit.refined.api.{RefType, Refined}
+import eu.timepit.refined.collection.NonEmpty
+import io.renku.tinytypes.{From, TinyType, TinyTypeFactory}
+import org.http4s.multipart.{Multipart, Part}
+import org.http4s.{DecodeFailure, DecodeResult, EntityDecoder, MalformedMessageBodyFailure}
+
+object syntax {
+
+  final implicit class EncoderOps[A](private val value: A) extends AnyVal {
+
+    def asPart[F[_]](partName: String)(implicit encoder: PartEncoder[F, A]): Part[F] =
+      encoder(partName, value)
+  }
+
+  final implicit class EncoderFOps[M[_]: Functor, A](private val value: M[A]) {
+
+    def asParts[F[_]](partName: String)(implicit encoder: PartEncoder[F, A]): M[Part[F]] =
+      value.map(_.asPart[F](s"$partName$partNameSuffix"))
+
+    private lazy val partNameSuffix: String = value match {
+      case _: Iterable[_] => "[]"
+      case _ => ""
+    }
+  }
+
+  final implicit class MultipartOps[F[_]: MonadThrow](private val multipart: Multipart[F]) {
+
+    def part(partName: String): F[Part[F]] =
+      findPart(partName).fold(new Exception(s"No '$partName' in the request").raiseError[F, Part[F]])(_.pure[F])
+
+    def findPart(partName: String): Option[Part[F]] =
+      multipart.parts.find(_.name contains partName)
+
+    def findParts(partName: String): List[Part[F]] =
```

Comment:
Only for consistency with similar apis, I would expect a `find...` to return an `Option`. What do you think about `filterParts` as a name?

Reply:
Great idea!